### ISSUE TITLE
Can not determine epoch number as wanted

### ISSUE THREAD

**Issue by kadir-gunel** [marian-nmt/marian#69]
Hello,

I have difficulty to train my model as I wanted to be. 

Case # 1: I set epoch size as 90 with `--after-epoch` flag. And I see in the train.log file that epoch size is 90. But interestingly train operation finishes right after 60th epoch. Why is this happening ? Did anyone experienced this ? 

Case # 2: Also, since training finishes early, I set `--overwrite` flag in the model file. Then it seems like the model continues to train from the last epoch. But when it starts to run validation script it shows much higher values for cross-entropy than the previous one. 

For case #1, I can provide log files. For case  #2, I need to check it because I might deleted. But still, I can reproduce it.

B.R.

p.s. This post seems like 2 separated issues but they are linked so I wrote as a single post.

**Comment by geovedi** [comment]
**on early stopped training**: was the `early-stopping` value set to default? you could check `valid.log` file, to see if there's how many time the training process was stalled. if it's 10, then it's the expected behaviour.

**on continued training result**: i'm having similar issue, doing some test now. see my discussion with @emjotde last night [here](https://groups.google.com/d/topic/amunmt/G0ZSe-BLB-Q/discussion).

**Comment by kadir-gunel** [comment]
yes, `--early-stopping` was set to 5. So, basically it assumes that there is no improvement further, right ?

**Comment by emjotde** [comment]
Hi,
@1: Right, that will be early stopping if you have a validation set. You can disable it with `early-stopping: 0`. BTW, 90 epochs? This seems unusual. 

@2: I suppose you have `moving-average: true` set. This is a fresh feature and I have seen it randomly cause trouble with restarting. Maybe disable it.

**Comment by emjotde** [comment]
early-stopping checks your performance on a validation set. Either with valid-sets or with valid-script-path. If there is no improvement for the first metric specified for 5 consecutive validations, it will stop training. Usually that's a good thing, but you might want to tune the parameter.

**Comment by kadir-gunel** [comment]
case#1 @emjotde I increased drop-outs in order to see how much improvement/diminishment I can get. 

case #2 @emjotde yes, also `--moving-average` was set to true. Ok, I will try again without it.

thanks for clarification of early-stopping. :+1:

**Comment by geovedi** [comment]
@kadir-gunel i just got the first iteration result from my resumed training without `moving-average` set, got some improvement on the result. but i didn't set `overwrite`, though.

**Comment by emjotde** [comment]
`overwrite` only causes the model.iterxxxxxx.npz models not to be saved. OK, that means I need to take a look at the moving-average problem. Probably save a separate unmodified model for resuming or something.

**Comment by kadir-gunel** [comment]
@geovedi I've already started a fresh experiment without `moving-average.` But I'll do it with a "trained" model tomorrow for sure. 
@emjotde so, you mean `overwrite` has nothing to do with continuation ?

**Comment by emjotde** [comment]
Hm, it should not, the training should continue by loading model.npz, this model is being saved regardless of `overwrite`. It should just stop creating the per-iteration models.

**Comment by kadir-gunel** [comment]
Interesting, because without `overwrite` flag I could not continue to train.

**Comment by emjotde** [comment]
oh? OK, then I need to check this too.

**Comment by geovedi** [comment]
@kadir-gunel  i can resume training without `overwrite`. what commit version are you using? mine is using 0fa1187.

**Comment by kadir-gunel** [comment]
it is master.

**Comment by emjotde** [comment]
Oh, wait, you sayid**without** `overwrite`. No, that should work without that flag. Using it all the time.

**Comment by emjotde** [comment]
No further comments here, assuming it's resolved.
